tests/rptest/services/redpanda.py

```diff
@@ -1715,6 +1715,19 @@ def pids(self, node):
     def started_nodes(self):
         return self._started
 
+    def render(self, path, **kwargs):
+        # Mitigate a race between checking for template_loader and template_env
+        # inside TemplateRenderer::render
+        assert (path is not None)
+        if hasattr(self, 'template_loader'):
+            wait_until(
```

Comment:
I think it would be neater to do just put a lock around calls to self.render.  That's kind of ugly too, but at least explicitly makes clear that we're dealing with a piece of thread-unsafe code.

a `with self.config_file_lock:` around self.render calls should do the trick

...that existing lock is dead code but conveniently named by total coincidence! :zany_face: